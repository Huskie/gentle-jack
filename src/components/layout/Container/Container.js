import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'react-emotion';

const StyledContainer = Styled('div')`
    margin: 0 auto;
    max-width: ${props => props.theme.sizing.maxWidth};
    padding: 0 24px;
`;

const Container = ({ children }) => <StyledContainer>{children}</StyledContainer>;

Container.propTypes = {
    children: PropTypes.node
};

Container.defaultProps = {};

export { Container };
