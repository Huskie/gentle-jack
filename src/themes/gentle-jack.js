const theme = {
  borderRadius: '4px',
  colours: {
    alpha: '#2742b8',
    beta: '#222',
    gamma: '#06f',
    delta: '#fff',
    epsilon: '#1b6cff',
    zeta: '#ffd561',
    eta: '#4048ef',
  },
  gradients: {
    alpha: '#4048ef, #2742b8',
  },
  spacing: {
    componentMargin: '16px',
  },
  sizing: {
    maxWidth: '1080px',
  },
};

export { theme };
