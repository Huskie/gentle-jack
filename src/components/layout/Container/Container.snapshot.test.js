import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'emotion-theming';

import { Container } from './Container';
import { theme } from '../../../themes/gentle-jack';

describe('Container snapshots:', () => {
  it('component renders with defaults', () => {
    const tree = renderer.create(
      <ThemeProvider theme={theme}>
        <Container>
          <p>A test child.</p>
          <span>Another test child.</span>
        </Container>
      </ThemeProvider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
