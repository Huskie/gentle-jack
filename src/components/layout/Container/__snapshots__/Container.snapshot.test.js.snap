// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container snapshots: component renders with defaults 1`] = `
.emotion-0 {
  margin: 0 auto;
  max-width: 1080px;
  padding: 0 24px;
}

<div
  className="emotion-0"
>
  <p>
    A test child.
  </p>
  <span>
    Another test child.
  </span>
</div>
`;
